final_project.py

```diff
@@ -0,0 +1,369 @@
+# 
+# Arguments: 
+#   string_input: block of text containing the network information
+#
+#   You may assume that for all the test cases we will use, you will be given the 
+#   connections and games liked for all users listed on the right-hand side of an
+#   'is connected to' statement. For example, we will not use the string 
+#   "A is connected to B.A likes to play X, Y, Z.C is connected to A.C likes to play X."
+#   as a test case for create_data_structure because the string does not 
+#   list B's connections or liked games.
+#   
+#   The procedure should be able to handle an empty string (the string '') as input, in
+#   which case it should return a network with no users.
+# 
+# Return:
+#   The newly created network data structure
+def create_data_structure(s):
+    network = {} # {user1:[[connections],[games]], user2:[[connections],[games]]...} 
+    user = []
+    connection = []
+    game = []
+    string_ls = s.split('.')
+    #print string_ls
+    if len(s) > 1:
+        
+        for e in string_ls:
+        
+            if 'is connected to' in e:
+                s = e[e.find('is connected to') + 16:] #s is the names of the
+                #connection and the next string to be used
+                #print s
+                s = s.split(', ')
+                user_name = e[:e.find(' is connected to')]
+                user.append(user_name)
+                connection.append(s)
+                #print connection, user
+            
+            if 'likes to play' in e:
+                s = e[e.find('likes to play') + 14:] # "+13" will leave a space
+                #before the game title, e.g. " The Legend of Corgi".
+                s = s.split(', ')
+                game.append(s)
+                #print game
+            else:
+                #return []
+                pass
+        n_users = len(user)
+        for i in range(n_users): #11 is the len of the user, connection, game list
+
+            network[user[i]] = connection[i], game[i]
+
+        #print network
+        return network
+    else:
+        return {}
+# ----------------------------------------------------------------------------- # 
+# Note that the first argument to all procedures below is 'network' This is the #
+# data structure that you created with your create_data_structure procedure,    #
+# though it may be modified as you add new users or new connections. Each       #
+# procedure below will then modify or extract information from 'network'        # 
+# ----------------------------------------------------------------------------- #
+
+# ----------------------------------------------------------------------------- 
+# get_connections(network, user): 
+#   Returns a list of all the connections that user has
+#
+# Arguments: 
+#   network: the gamer network data structure
+#   user:    a string containing the name of the user
+# 
+# Return: 
+#   A list of all connections the user has.
+#   - If the user has no connections, return an empty list.
```

Comment:
does your code handle this case? when the user is in the network but has no connections ?